frontend/src/stores/feature-flag.ts

```diff
@@ -131,8 +131,23 @@ const FEATURE_FLAG = "feature_flag"
 
 export const useFeatureFlagStore = defineStore(FEATURE_FLAG, {
   state: () => {
-    const deploymentEnv = useRuntimeConfig().public.deploymentEnv as DeployEnv
-    return initializeFlagState(deploymentEnv)
+    const nuxtApp = tryUseNuxtApp()
+    if (nuxtApp) {
+      return initializeFlagState(
+        nuxtApp.$config.public.deploymentEnv as DeployEnv
+      )
+    }
+
+    const deploymentEnv = process.env.NUXT_PUBLIC_DEPLOYMENT_ENV as DeployEnv
```

Comment:
I guess this is _technically_ safe, because as you said, this loss of context occurs on the server, so client code will never reach this line... but isn't `import.meta.env` the correct way to read an environment variable, for idempotent code (which this technically is, as it isn't explicitly scoped)?

I also wonder, after reading the Nuxt docs, whether this fallback approach actually paints over a deeper problem of misused composables and asynchronous code. Based on the Nuxt docs you linked, it seemed like the real fix was to prevent this code running in an unsupported context, rather than papering over the issue with this fallback.

Is the fallback still necessary after the other refactors you made to remove the deep nesting? It would be nice to do that refactor first, and then see if the problem still happens, and if so trying and find the root cause rather than a band-aid fix. At least, that's if I've understood the docs correctly when it comes to what causes this lack of Nuxt context in certain code paths.

Reply:
> Is the fallback still necessary after the other refactors you made to remove the deep nesting?

Debugging this is kind of tricky since this happens so rarely. But from what I can see, we _**don't need**_ this anymore. Before removing the deep nesting, we had 2-3 test failures per Playwright test suite run. Now, after adding these changes, and rebasing another PR onto this one, I don't see the failures AND there are no logs that would happen if nuxt content wasn't available in the store.